pkg/api/vlabs/types.go

```diff
@@ -30,6 +30,7 @@ type ContainerService struct {
 
 // Properties represents the ACS cluster definition
 type Properties struct {
+	Name                    string                   `json:"clusterName,omitempty"`
```

Comment:
What happens to the resource names if the user doesn't provide a cluster name?

Reply:
Good question, Curretnly It should fail I guess because I did implement but I guess it should default to the old `<orchestrator>-<########>-...` name.

Do you agree ?

Reply:
It won't really be like before cause you'll have `<orchestrator>-<########>-master-1` instead of `<orchestrator>-master-<########>--1` but it's pretty close.

Reply:
It'd be good to keep the exact same name as before for users who don't provide a cluster name I think. That way we don't introduce change for people who have in place tools that rely on the existing name and using the new naming is optional.

Reply:
Seems difficult as I'm modifying the templates to be like `<cluster_name>-<type>-<#>[-...]`.

What I can probably do is to have `<cluster_name>` default to `<orchestrator>-<########>` if not provided.

Reply:
Let's make sure that naming resources stays the same if clusterName is not provided by the user to ensure backward compatibility